src/controller/CHIPDeviceController.cpp

```diff
@@ -1742,6 +1745,9 @@ void DeviceCommissioner::OnDone(app::ReadClient *)
                 return CHIP_NO_ERROR;
             }
 
+            // this code is checking if the device is already on the commissioner's fabric.
+            // if a matching fabric is found, then remember the nodeId so that the commissioner
+            // can cancel commissioning (before it fails in AddNoc) and know it's nodeId.
```

Comment:
Please add an integration test for this. This is a late change to AutoCommissioner logic used by *many* participants. This cannot be changed like this without an integration test validating the behavior. An integration test can be constructed using Python ChipDeviceController and run as part of REPL tests. You would commission a first fabric, then commission a node on same fabric with autocommissioning, and try to see that it doesn't fail.

Reply:
Updated to make this functionality runtime enable-able (disabled by default)